tools/ping/src/lib.rs

```diff
@@ -312,6 +325,23 @@ impl AppInfo {
     fn peer_id_to_account_id(&self, peer_id: &PeerId) -> Option<&AccountId> {
         self.stats.get(peer_id).and_then(|s| s.account_id.as_ref())
     }
+
+    fn push_metrics(&self) {
+        let metric_families = prometheus::gather();
+        if let Some(metrics_gateway_address) = &self.metrics_gateway_address {
+            prometheus::push_metrics(
```

Comment:
so I tested this and it panics :(

```
2022-11-09T19:35:55.210661Z  INFO neard: version="trunk" build="1.1.0-3480-g70c3f2036" latest_protocol=57
2022-11-09T19:35:55.210803Z  WARN ping: the ping command is not stable, and may be removed or changed arbitrarily at any time
2022-11-09T19:35:55.214209Z  INFO network: Connection to ed25519:7PGseFbWxvYVgZ89K1uTJKYoKetWs7BJtbyXDzfbAcqX@127.0.0.1:24567 established. latency: 368µs277ns
2022-11-09T19:35:55.224317Z DEBUG network: received PeerMessage::Handshake len: 227
2022-11-09T19:35:55.224412Z  INFO network: handshake latency: 8ms864µs728ns
2022-11-09T19:35:55.224748Z DEBUG network: received PeerMessage::SyncRoutingTable len: 715
2022-11-09T19:35:55.225382Z DEBUG network: received PeerMessage::PeersRequest len: 2
thread 'main' panicked at 'Cannot drop a runtime in a context where blocking is not allowed. This happens when a runtime is dropped from within an asynchronous context.', /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/blocking/shutdown.rs:51:21
stack backtrace:
   0: std::panicking::begin_panic
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:616:12
   1: tokio::runtime::blocking::shutdown::Receiver::wait
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/blocking/shutdown.rs:51:21
   2: tokio::runtime::blocking::pool::BlockingPool::shutdown
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/blocking/pool.rs:193:12
   3: <tokio::runtime::blocking::pool::BlockingPool as core::ops::drop::Drop>::drop
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/blocking/pool.rs:210:9
   4: core::ptr::drop_in_place<tokio::runtime::blocking::pool::BlockingPool>
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ptr/mod.rs:487:1
   5: core::ptr::drop_in_place<tokio::runtime::Runtime>
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ptr/mod.rs:487:1
   6: reqwest::blocking::wait::enter
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/reqwest-0.11.10/src/blocking/wait.rs:76:21
   7: reqwest::blocking::wait::timeout
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/reqwest-0.11.10/src/blocking/wait.rs:13:5
   8: reqwest::blocking::client::ClientHandle::new
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/reqwest-0.11.10/src/blocking/client.rs:1001:15
   9: reqwest::blocking::client::ClientBuilder::build
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/reqwest-0.11.10/src/blocking/client.rs:103:9
  10: <prometheus::push::HTTP_CLIENT as core::ops::deref::Deref>::deref::__static_ref_initialize
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/prometheus-0.13.1/src/push.rs:24:38
  11: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
  12: spin::once::Once<T>::call_once
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/spin-0.5.2/src/once.rs:110:50
  13: lazy_static::lazy::Lazy<T>::get
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/core_lazy.rs:21:9
  14: <prometheus::push::HTTP_CLIENT as core::ops::deref::Deref>::deref::__stability
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/lib.rs:142:21
  15: <prometheus::push::HTTP_CLIENT as core::ops::deref::Deref>::deref
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/lib.rs:144:17
  16: prometheus::push::push
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/prometheus-0.13.1/src/push.rs:151:23
  17: prometheus::push::push_metrics
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/prometheus-0.13.1/src/push.rs:62:5
  18: near_ping::AppInfo::push_metrics
             at ./tools/ping/src/lib.rs:332:13
  19: near_ping::AppInfo::ping_sent
             at ./tools/ping/src/lib.rs:161:9
  20: near_ping::ping_via_node::{{closure}}
             at ./tools/ping/src/lib.rs:473:17
  21: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/future/mod.rs:91:19
  22: near_ping::cli::PingCommand::run::{{closure}}
             at ./tools/ping/src/cli.rs:220:13
  23: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/future/mod.rs:91:19
  24: tokio::park::thread::CachedParkThread::block_on::{{closure}}
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/park/thread.rs:263:54
  25: tokio::coop::with_budget::{{closure}}
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/coop.rs:102:9
  26: std::thread::local::LocalKey<T>::try_with
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/thread/local.rs:445:16
  27: std::thread::local::LocalKey<T>::with
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/thread/local.rs:421:9
  28: tokio::coop::with_budget
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/coop.rs:95:5
  29: tokio::coop::budget
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/coop.rs:72:5
  30: tokio::park::thread::CachedParkThread::block_on
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/park/thread.rs:263:31
  31: tokio::runtime::enter::Enter::block_on
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/enter.rs:151:13
  32: tokio::runtime::thread_pool::ThreadPool::block_on
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/thread_pool/mod.rs:81:9
  33: tokio::runtime::Runtime::block_on
             at /home/marcelo/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/mod.rs:477:43
  34: near_ping::cli::PingCommand::run
             at ./tools/ping/src/cli.rs:203:9
  35: neard::cli::NeardCmd::parse_and_run
             at ./neard/src/cli.rs:104:17
  36: neard::main
             at ./neard/src/main.rs:62:5
  37: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Aborted
```

I found this issue which seems like the same problem: https://github.com/tikv/rust-prometheus/issues/453. honestly not sure what to do about that, since it seems like `push_metrics()` just doesnt work if youre using tokio. Do we need to use `push_metrics()`? Maybe it will be fine to just have the job scraped as usual until that issue is fixed?